consensus/src/round_manager.rs

```diff
@@ -316,6 +322,21 @@ impl RoundManager {
         }
     }
 
+    pub async fn process_verified_self_proposal_msg(
+        &mut self,
+        proposal: Block,
+    ) -> anyhow::Result<()> {
+        if proposal.round() < self.round_state.current_round() {
```

Comment:
this should be !=

Reply:
Can you clarify why is that the case? The round should be strictly in increasing order right?

Reply:
the round can increase in the mean time by other messages

Reply:
Yes if the `self.round_state.current_round()` is strictly increasing, this check should be fine, right? or Am I missing something?

Reply:
oh I guess that's true, but it also doesn't hurt to just put a `ensure!(round == self.round)`

Reply:
Yes, sure makes sense. Just wanted to confirm my understanding is correct.